### PR TITLE
[Bug](runtime-filter) fix wrong build_bf_exactly when sync filter size disabled

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
### What problem does this PR solve?
fix wrong build_bf_exactly when sync filter size disabled

introduced by #44169
### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/exprs/runtime_filter.cpp [hunk 1]
```diff
@@ -1366,21 +1365,10 @@ Status IRuntimeFilter::init_with_desc(const TRuntimeFilterDesc* desc, const TQue
...
-    auto sync_filter_size = desc->__isset.sync_filter_size && desc->sync_filter_size;
-    // We build runtime filter by exact distinct count if all of 3 conditions are met:
-    // 1. Only 1 join key
-    // 2. Bloom filter
-    // 3. Size of all bloom filters will be same (size will be sync or this is a broadcast join).
-    params.build_bf_exactly =
-            build_bf_exactly && (_runtime_filter_type == RuntimeFilterType::BLOOM_FILTER ||
-                                 _runtime_filter_type == RuntimeFilterType::IN_OR_BLOOM_FILTER);
 
+    params.build_bf_exactly = desc->build_bf_exactly;
```
Comment (Gabriel39) [inline be/src/exprs/runtime_filter.cpp:1369]:
```suggestion
    params.build_bf_exactly = desc->__isset.build_bf_exactly && desc->build_bf_exactly;
```

File: fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilter.java [hunk 2]
```diff
@@ -597,6 +624,14 @@ public void addTarget(RuntimeFilterTarget target) {
         targets.add(target);
     }
 
+    public void setSingleEq(int eqJoinConjunctsNumbers) {
+        singleEq = (eqJoinConjunctsNumbers == 1);
+    }
+
+    public boolean getSingleEq() {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilter.java:631]: unused?

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilter.java [hunk 3]
```diff
@@ -216,9 +218,36 @@ public TRuntimeFilterDesc toThrift() {
...
         for (RuntimeFilterTarget target : targets) {
             tFilter.putToPlanIdToTargetExpr(target.node.getId().asInt(), target.expr.treeToThrift());
+            hasSerialTargets = hasSerialTargets
+                    || (target.node.isSerialOperator() && target.node.fragment.useSerialSource(ConnectContext.get()));
         }
+
+        boolean enableSyncFilterSize = ConnectContext.get() != null
+                && ConnectContext.get().getSessionVariable().enableSyncRuntimeFilterSize();
+        // there are two cases need merge rf
+        // 1. hasRemoteTargets is true means join type is hash shuffle join then rf will
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilter.java:232]: `hasRemoteTargets` not means `shuffle join`, maybe `broadcast`

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**